Updated dependencies + code to Purescript 0.9

PipocaQuemada opened the pull request.

Commit:
Updated dependencies + code to Purescript 0.9

hdgarrood commented on src/Data/Version/Haskell.purs:
```diff
@@ -9,24 +9,24 @@
 module Data.Version.Haskell where
 
-import Prelude
+import Prelude (class Show, class Ord, class Eq, show, compare, flip, pure, bind, map, (<>), (==), (&&), (<<<), ($), (*>), (||), (<$>))
```
Can we keep the Prelude import implicit here? IIRC you're allowed one implicit import before you start getting warnings.

hdgarrood commented:
It looks like the CI is failing because of an old version of either Pulp or purescript (I haven't looked closely). Also, we should stop testing on node.js versions <4 at the same time, as I don't think Pulp supports them any more, and I'm fairly sure quite a few other parts of the PS ecosystem no longer do either.

Other than that, looks great, thanks!

Commit:
updated pulp's version; removed explicit import list from 'import prelude'.

Commit:
Fixed travis build.

PipocaQuemada commented:
I removed the old versions of node, and fixed up travis to install and use bower.

Currently, it's timing out after getting to

> pulp build
> - Building project in /home/travis/build/hdgarrood/purescript-versions

Since nothing's printed for the next 10 minutes.  That seems rather long for compilation to take; any idea of what's going wrong?

Commit:
Print out version information in travis build.

hdgarrood commented:
I think this is something we've seen a few times before, we think the cause is that Travis reports that it has more cores available than it actually does. Try adding `dist: trusty` and `sudo: required` to .travis.yml. You could look at prelude's .travis.yml as an example.

Commit:
Use trusty in travis.

PipocaQuemada commented:
Looks like travis builds, now.  Thanks!

hdgarrood commented:
Great, thanks!

hdgarrood commented:
I've just tagged 2.0.0. Could you please check everything works and publish to Pursuit, if you can? I don't actually have access to a functioning PC at the moment.
